mgwr/utils.py

```diff
@@ -59,61 +69,97 @@ def shift_colormap(cmap, start=0, midpoint=0.5, stop=1.0, name='shiftedcmap'):
-    n : Number of intervals between the min and max values for the gradient of the truncated colormap
-          
-    Returns
-    -------
-    new_cmap : A new colormap that has been shifted. 
+    Args:
+        cmap (LinearSegmentedColormap) : Matplotlib colormap to be altered
+        min_val (float) : Minimum value of the original colormap to include in the truncated colormap
+        max_val (float) : Maximum value of the original colormap to include in the truncated colormap
+        n (int) : Number of intervals between the min and max values for the gradient of the truncated colormap
+
+    Returns:
+        new_camap (LinearSegmentedColormap) : A new colormap that has been truncated.
+
+    Examples:
+        >>> import numpy as np
+        >>> import matplotlib.pyplot as plt
+        >>> from mgwr.utils import truncate_colormap
+        >>> cmap = plt.cm.seismic
+        >>> cmap = truncate_colormap(cmap, min_val=0.0, max_val=0.5, n=100)
+
     '''
 
     import matplotlib as mpl
 
     new_cmap = mpl.colors.LinearSegmentedColormap.from_list(
-        'trunc({n},{a:.2f},{b:.2f})'.format(n=cmap.name, a=minval, b=maxval),
-        cmap(np.linspace(minval, maxval, n)))
+        'trunc({n},{a:.2f},{b:.2f})'.format(n=cmap.name, a=min_val, b=max_val),
+        cmap(np.linspace(min_val, max_val, n)))
     return new_cmap
 
 
 @requires('matplotlib')
 @requires('geopandas')
-def compare_surfaces(data, var1, var2, gwr_t, gwr_bw, mgwr_t, mgwr_bw, name,
-                     kwargs1, kwargs2, savefig=None):
+def compare_surfaces(data: pd.DataFrame,
+                     var1: str,
+                     var2: str,
+                     gwr_t: str,
+                     gwr_bw: float,
+                     mgwr_t: str,
+                     mgwr_bw: float,
+                     name: str,
+                     kwargs1,
+                     kwargs2,
+                     savefig: str = None) -> None:
     '''
     Function that creates comparative visualization of GWR and MGWR surfaces.
 
-    Parameters
-    ----------
-    data   : pandas or geopandas Dataframe
-             gwr/mgwr results
-    var1   : string
-             name of gwr parameter estimate column in frame
-    var2   : string
-             name of mgwr parameter estimate column in frame
-    gwr_t  : string
-             name of gwr t-values column in frame associated with var1
-    gwr_bw : float
-             bandwidth for gwr model for var1
-    mgwr_t : string
-             name of mgwr t-values column in frame associated with var2
-    mgwr_bw: float
-             bandwidth for mgwr model for var2
-    name   : string
-             common variable name to use for title
-    kwargs1:
-             additional plotting arguments for gwr surface
-    kwargs2:
-             additional plotting arguments for mgwr surface
-    savefig: string, optional
-             path to save the figure. Default is None. Not to save figure.
+    Args:
```

Comment:
You are breaking numpydoc style again

Reply:
Hi Martin,

Thank you for your recommendation. I understand that Sphinx accepts both NumPyDoc and Google Docstring styles. Given Google Docstring's broader acceptance, I've been using it for new functions. From your comments, it seems you're advocating for consistency in existing documentation style. Acknowledging your point, I will also adhere to the NumPyDoc style to maintain this consistency.